tests/hil/tests/settings/test.c

```diff
@@ -2,11 +2,14 @@
 #include <golioth/golioth_debug.h>
 #include <golioth/golioth_sys.h>
 #include <golioth/settings.h>
+#include <string.h>
 
 LOG_TAG_DEFINE(test_settings);
 
 static golioth_sys_sem_t connected_sem;
 static golioth_sys_sem_t cancel_all_sem;
+static golioth_sys_sem_t restart_sem;
+static golioth_sys_sem_t restart_test_flag_sem;
```

Comment:
This is pretty confusing - I think I was able to figure it out eventually, but it's immediately not clear why we're using 2 semaphores, and their names aren't descriptive enough to distinguish them from each other. This is so that the test doesn't get re-triggered when observations are re-established, correct? Can we handle this the same way we do for the cancel observations test? By using a boolean and setting it back to false before the client reconnects. We should also confirm that the device is disconnected by attempting to change a setting, again similar to the cancel_all test.

Reply:
The problem I ran into in that approach is that the device can't be updated while the client is stopped. So the alternative to what I've done here is to stop the client and wait for an arbitrary amount of time, then start it again. Do you prefer that?